src/CopyrighFixer/CopyrighFixer/sign.cpp

```diff
@@ -37,4 +45,115 @@ const QString &Signature::getMessage() const {
     return _customMessage;
 }
 
+bool Signature::fromJson(const QString &pathToFile) {
+
+    QFileInfo signFile(pathToFile);
+    if (signFile.exists() && signFile.isFile()) {
+
+        QFile file(pathToFile);
+        if (!file.open(QIODevice::ReadOnly)) {
+            QuasarAppUtils::Params::log("Json file couldn't be opened/found");
+            return false;
+        }
+
+        QByteArray byteSignFile(file.readAll());
+        file.close();
+
+        QJsonDocument jsDoc(QJsonDocument::fromJson(byteSignFile));
+        if (jsDoc.isNull()) {
+            QuasarAppUtils::Params::log("Failed to create JSON doc.");
+            return false;
+        }
+        if (!jsDoc.isObject()) {
+            QuasarAppUtils::Params::log("JSON is not an object.");
+            return false;
+        }
+
+        QJsonObject jsObj = jsDoc.object();
+        if (jsObj.isEmpty()) {
+            QuasarAppUtils::Params::log("JSON object is empty.");
+            return false;
+        }
+
+        Owner ownObj;
+        QJsonArray ownLst(jsObj.value("ownersList").toArray());
+        for (auto itemLst = ownLst.cbegin(); itemLst != ownLst.end(); ++itemLst) {
+            ownObj.fromjson(itemLst->toObject());
+            _ownersMap.insert(itemLst->toObject().value("timePoint").toInt(), ownObj);
+        }
+
+        _licenseTitle = jsObj.value("license").toString();
+        _customMessage = jsObj.value("licenseText").toString();
+
+        return true;
+
+    } else {
+        return false;
+    }
+
+}
+
+bool Signature::toJson(QString &pathToFile) const {
```

Comment:
use the const reference please.
```suggestion
bool Signature::toJson(const QString &pathToFile) const {
```